File: src/main/java/se/oeman/SensorActor.java
package se.oeman;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;

public class SensorActor extends UntypedActor {
  SensorsJNACallback sensorsjnacallback;

  public SensorActor(ActorRef ref) {
    sensorsjnacallback = new SensorsJNACallback(ref, getSelf());
  }

  @Override
  public void preStart() {
    sensorsjnacallback.startListening();
  }

  @Override
  public void postStop() {
    sensorsjnacallback.stopListening();
  }

  @Override
  public void onReceive(Object message) throws Exception {
    System.out.println(message);
  }
}

File: src/main/scala/se/oeman/SensorActor.scala
package se.oeman

import akka.actor.{Actor, ActorRef}


class SensorActor(ref: ActorRef) extends Actor {
  val sensorsjnacallback = new SensorsJNACallback(ref, self)

  override def preStart = {
    println("Starting sensor listener")
    sensorsjnacallback.startListening()
  }

  override def postStop = sensorsjnacallback.stopListening()

  override def receive = {
    case x => println(s"What?: $x")
  }
}
